panic after edit-finality-provider

Hello, I just ran a `fpd edit-finality-provider xxxx --commission-rate 0.07` and the daemon panicked with the following trace

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2534783]

goroutine 15158 [running]:
github.com/babylonlabs-io/finality-provider/finality-provider/store.(*FinalityProviderStore).SetFpDescription.func1(0xc0028b90e0)
	/build/finality-provider/store/fpstore.go:273 +0x163
github.com/babylonlabs-io/finality-provider/finality-provider/store.(*FinalityProviderStore).setFinalityProviderState.func1({0x4dacc38?, 0xc00189a8a8?})
	/build/finality-provider/store/fpstore.go:182 +0xee
github.com/btcsuite/btcwallet/walletdb/bdb.(*db).Batch.func1(0xc0022f00e0)
	/go/pkg/mod/github.com/btcsuite/btcwallet/walletdb@v1.4.0/bdb/db.go:364 +0x5c
go.etcd.io/bbolt.(*DB).Update(0x0?, 0xc001bd8550)
	/go/pkg/mod/go.etcd.io/bbolt@v1.4.0-alpha.0.0.20240404170359-43604f3112c5/db.go:924 +0x6f
go.etcd.io/bbolt.(*DB).Batch(0xc00156afc8, 0xc001bd8550)
	/go/pkg/mod/go.etcd.io/bbolt@v1.4.0-alpha.0.0.20240404170359-43604f3112c5/db.go:1002 +0x32b
github.com/btcsuite/btcwallet/walletdb/bdb.(*db).Batch(0xc00156afc8, 0xc00206e870)
	/go/pkg/mod/github.com/btcsuite/btcwallet/walletdb@v1.4.0/bdb/db.go:361 +0x5e
github.com/btcsuite/btcwallet/walletdb.Batch({0x4da93f0?, 0xc00156afc8?}, 0xc00206e870?)
	/go/pkg/mod/github.com/btcsuite/btcwallet/walletdb@v1.4.0/interface.go:290 +0x4a
github.com/lightningnetwork/lnd/kvdb.Batch({0x4da93f0?, 0xc00156afc8?}, 0xc00206e870?)
	/go/pkg/mod/github.com/lightningnetwork/lnd/kvdb@v1.4.1/interface.go:44 +0x4d
github.com/babylonlabs-io/finality-provider/finality-provider/store.(*FinalityProviderStore).setFinalityProviderState(0xc00065d400, 0xc001624ba0?, 0xc001aa2300)
	/build/finality-provider/store/fpstore.go:166 +0x136
github.com/babylonlabs-io/finality-provider/finality-provider/store.(*FinalityProviderStore).SetFpDescription(0xc00065d400, 0xc001c32190, 0xc001e251d0, 0xc00190c1b0)
	/build/finality-provider/store/fpstore.go:279 +0x7e
github.com/babylonlabs-io/finality-provider/finality-provider/service.(*rpcServer).EditFinalityProvider(0xc00268dc80, {0x4204d00?, 0xc0000e68b8?}, 0xc0020bb0e0)
	/build/finality-provider/service/rpcserver.go:231 +0x18d
github.com/babylonlabs-io/finality-provider/finality-provider/proto._FinalityProviders_EditFinalityProvider_Handler({0x4204d00, 0xc00268dc80}, {0x4d99250, 0xc001491320}, 0xc00155aa00, 0x0)
	/build/finality-provider/proto/finality_providers_grpc.pb.go:335 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0026b8200, {0x4d99250, 0xc002c6a6c0}, 0xc0020b9260, 0xc0029962a0, 0x72da5d0, 0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.70.0/server.go:1400 +0x103b
google.golang.org/grpc.(*Server).handleStream(0xc0026b8200, {0x4d99c78, 0xc0028c6000}, 0xc0020b9260)
	/go/pkg/mod/google.golang.org/grpc@v1.70.0/server.go:1810 +0xbaa
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	/go/pkg/mod/google.golang.org/grpc@v1.70.0/server.go:1030 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 15129
	/go/pkg/mod/google.golang.org/grpc@v1.70.0/server.go:1041 +0x125
```

Current version:
```
$ fpd version
Version:       1.0.0-rc.4-hot-fix
Git Commit:    cdc6e31
Git Timestamp: 2025-03-26T22:44:10Z
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -52,6 +52,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 * [#399](https://github.com/babylonlabs-io/finality-provider/pull/399) chore: submit immediately
 * [#410](https://github.com/babylonlabs-io/finality-provider/pull/410) chore: rollback sign records from eots store
 * [#413](https://github.com/babylonlabs-io/finality-provider/pull/413) chore: rm default value for app hash flag
+* [#427](https://github.com/babylonlabs-io/finality-provider/pull/427) fix: handle uninitialized commissionInfo when editing fp
 
 ## v1.0.0-rc.2
 

--- a/finality-provider/store/fpstore.go
+++ b/finality-provider/store/fpstore.go
@@ -270,6 +270,9 @@ func (s *FinalityProviderStore) SetFpDescription(btcPk *btcec.PublicKey, desc *s
 		fp.Description = descBytes
 		if rate != nil {
 			fp.Commission = rate.String()
+			if fp.CommissionInfo == nil {
+				fp.CommissionInfo = &proto.CommissionInfo{}
+			}
 			fp.CommissionInfo.UpdateTime = timestamppb.New(time.Now().UTC())
 		}
 

